cmd/connect.go

```diff
@@ -72,11 +74,28 @@ func (o *ConnectOptions) Complete(f Factory, args []string) (err error) {
 			return err
 		}
 	} else if !QRIRepoInitialized(qriPath) {
-		return errors.New(repo.ErrNoRepo, "no qri repo exists\nhave you run 'qri setup'?")
+		return qerr.New(repo.ErrNoRepo, "no qri repo exists\nhave you run 'qri setup'?")
 	}
 
 	if err = f.Init(); err != nil {
-		return err
+		if errors.Is(err, qipfs.ErrNeedMigration) {
+			err = nil
+			msg := `
+Your IPFS repo needs updating before qri can start. 
+Run migration now?`
+			if !confirm(o.Out, o.In, msg, false) {
+				return qerr.New(qipfs.ErrNeedMigration, `connect cancelled, requires migration.`)
+			}
+
+			if err = qipfs.Migrate(); err != nil {
```

Comment:
Not sure if we should trigger the migration here, especially as we will have other migration logic as well. Currently migration logic is trggered at https://github.com/qri-io/qri/blob/4159a259b679b70ffe374c3f2ae62f2b3852508a/lib/lib.go#L187
And runs https://github.com/qri-io/qri/blob/master/config/migrate/migrate.go which should "automagically" happen before we get to this point if I'm not mistaken.

Reply:
Yea this is hold over/experimentation from before the `OneToTwo` function was planned.